include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -678,24 +693,41 @@ class FEEvaluationBase
   void read_dof_values_plain (const VectorType *src_data[]);
 
   /**
-   * Internal data fields that store the values. Derived classes will know the
-   * length of all arrays at compile time and allocate the memory on the
-   * stack. This makes it possible to cheaply set up a FEEvaluation object and
-   * write thread-safe programs by letting each thread own a private object of
-   * this type. In this base class, only pointers to the actual data are
-   * stored.
-   *
+   * This is the general array for all data fields.
+   */
+  AlignedVector<VectorizedArray<Number> > *scratch_data_array;
```

Comment:
OK, thanks for the explanation. I guess I'm neither fond of bare `new/delete` nor of pointer members pointing to either data that has been statically allocated *or* dynamically allocated. Both just lends itself to memory leaks or memory corruption (if you accidentally delete something you don't own).

But that's all for a different time to clean up. No worries.